framework/Optimizers/gradients/CentralDifference.py

```diff
@@ -39,7 +39,7 @@ def getInputSpecification(cls):
           """
     return specs
 
-  def chooseEvaluationPoints(self, opt, stepSize):
+  def chooseEvaluationPoints(self, opt, stepSize, **kwargs):
```

Comment:
docstring? do we use this as "kwargs" or simply a constrain dict?

Reply:
FD uses the extra kwarg for constraints, I can replace kwargs with that